feat: add support for Equinix Metal Load Balancer (#470)
kubernetes-sigs/cloud-provider-equinix-metal: README.md

| Kubernetes Service annotation to set EIP metro                                                                                                               |                | `METAL_ANNOTATION_EIP_METRO`            | `annotationEIPMetro`           | `"metal.equinix.com/eip-metro"`                              |
| Kubernetes Service annotation to set EIP facility                                                                                                            |                | `METAL_ANNOTATION_EIP_FACILITY`         | `annotationEIPFacility`        | `"metal.equinix.com/eip-facility"`                           |
| Tag for control plane Elastic IP                                                                                                                             |                | `METAL_EIP_TAG`                         | `eipTag`                       | No control plane Elastic IP                                  |

> | ID for control plane Equinix Metal Load Balancer                                                                                                                             |                | `METAL_LOAD_BALANCER_ID`                         | `loadBalancerID`                       | No control plane Equinix Metal Load Balancer                                  |

Reviewer: I think this option would fit with how we offer arguments in `METAL_LOAD_BALANCER`, as used by [metallb](https://github.com/kubernetes-sigs/cloud-provider-equinix-metal/blob/45dde1a565767e17e4a515a092a46c5ee4f3e744/metal/loadbalancers.go#L81-L89):
* `METAL_LOAD_BALANCER=emlb://sv`
* `METAL_LOAD_BALANCER=emlb://?loadbalancerID=123`
* `METAL_LOAD_BALANCER=emlb://?tags=cpem`

Reviewer: I know there is no support for tags today.

I'm curious how well CPEM will handle the limitation of working with a single LoadBalancer and whether this is more of a development feature or if this is an argument that users will want in their clusters.

Reviewer: Alternatively, if new options are the preferred approach, perhaps with #472, we may want to expose arguments in the future for the metallb options only offered through `METAL_LOAD_BALANCER` query parameters today.

Using these flags and annotations, you can run the CCM on a node in a different metro or facility, or even outside of Equinix Metal entirely.

> #### Service LoadBalancer Implementations

Reviewer: We should link to this section above in the table where the `loadbalancer` setting is introduced without any options described.

https://github.com/kubernetes-sigs/cloud-provider-equinix-metal/pull/470/commits/8a048dc5b3f9b735b49877bf41b8f15f265dfc7c#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R224

For loadbalancing for Kubernetes `Service` of `type=LoadBalancer`, the following implementations are supported:

- [Equinix Metal Load Balancer](#EquinixMetalLoadBalancer)
- [kube-vip](#kube-vip)
- [MetalLB](#metallb)
- [empty](#empty)

CCM does **not** deploy _any_ load balancers for you. It limits itself to managing the Equinix Metal-specific
API calls to support a load balancer, and providing configuration for supported load balancers.

##### Equinix Metal Load Balancer

> Equinix Metal Load Balancer (EMLB) is a beta service that is available to a limited number of Equinix Metal customers that provides managed layer 4 load balancers.

Reviewer: ```suggestion
[Equinix Metal Load Balancer](https://github.com/equinix/lbaas-api-docs) (EMLB) is a [service under development](https://feedback.equinixmetal.com/network/p/load-balancer-as-a-service-lbaas), currently in [beta](https://deploy.equinix.com/enroll-beta-lab/), that provides managed layer 4 load balancers.
```
